lewis/scripts/run.py

```
@@ -182,10 +189,11 @@ def run_simulation(argument_list=None):  # noqa: C901
             print('\n'.join(simulation_factory.get_protocols(arguments.device)))
             return
 
-        protocol, options = parse_adapter_options(arguments.adapter_options)
+        protocols = parse_adapter_options(
+            arguments.adapter_options) if not arguments.no_interface else {}
```

This looks really weird the way it's split.

Also, didn't you mention it should be an error if both `-p` and `-n` are specified? I think that's a good idea... better than silently ignoring `-p`. Don't forget release notes if you add this.

---

True, the line break makes this look a little weird.

I wasn't really sure about what is best in terms of behavior, so I tried this. But maybe it is indeed better to raise an error if both options are specified, it's a small change.